nsxt/resource_nsxt_policy_edge_transport_node.go

```diff
@@ -0,0 +1,1625 @@
+							Required:    true,
+							Sensitive:   true,
+							Description: "Node cli password",
+						},
+						"cli_username": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Default:     "admin",
+							Description: "CLI \"admin\" username",
+						},
+						"root_password": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Sensitive:   true,
+							Description: "Node root user password",
+						},
+					},
+				},
+			},
+			"failure_domain_path": {
+				Type:        schema.TypeString,
+				Description: "Path of the failure domain",
+				Optional:    true,
+			},
+			"form_factor": {
+				Type:         schema.TypeString,
+				Default:      model.PolicyEdgeTransportNode_FORM_FACTOR_MEDIUM,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(policyEdgeNodeFormFactorValues, false),
+			},
+			"hostname": {
+				Type:        schema.TypeString,
+				Description: "Host name or FQDN for edge node",
+				Required:    true,
+			},
+			"management_interface": {
+				Type:        schema.TypeList,
+				Description: "Applicable For LCM managed Node and contains the management interface info",
+				MaxItems:    1,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ip_assignment": getPolicyIPAssignmentSchema(true, 1, 2),
+						"network_id": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Portgroup, logical switch identifier or segment path for management network connectivity",
+						},
+					},
+				},
+			},
+			"switch": {
+				Type:        schema.TypeList,
+				Description: "Edge Transport Node switches configuration",
+				Required:    true,
+				MinItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"overlay_transport_zone_path": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "An overlay TransportZone path that is associated with the specified edge TN switch",
+						},
+						"pnic": {
+							Type:        schema.TypeList,
+							Required:    true,
+							Description: "Physical NIC specification",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"datapath_network_id": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "A portgroup, logical switch identifier or segment path for datapath connectivity",
+									},
+									"device_name": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "Device name or key e.g. fp-eth0, fp-eth1 etc",
+									},
+									"uplink_name": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "Uplink name for this Pnic. This name will be used to reference this Pnic in other configurations",
+									},
+								},
+							},
+						},
+						"uplink_host_switch_profile_path": {
```

Comment:
`validatePolicyPath` here and below?
Also those might be auto-assigned to default profile, in which case `Computed` is needed